### ISSUE TITLE
Continuous Song Cycling

### ISSUE THREAD

**Issue by kcrute** [mcrute/pydora#51]
When playing any station Pydora will continuously print song names out to the screen, 4 at a time, over and over. No music plays. I've only tried this on one computer: Mac running 10.12.3 and the latest version of Pydora.

**Comment by jcass77** [comment]
This sounds like a duplicate of #8.

It may be that pydora is connecting successfully to the Pandora servers (the server always just gives the next four songs to be played), but then a failure occurs when mpg123 tries to access and stream those files.

Have you tested mpg123 with anything else to verify that it has been installed successfully?

**Comment by mcrute** [comment]
This seems to be a somewhat recurring issue on certain versions of Mac OS (which I haven't been able to pin down precisely). mpg123 will hang but will continue to accept input from the player just never playing anything which will cause the playlist to cycle. I did some POC work to try to detect this but it was quite unreliable and I can no longer repro the issue. At the very least #8 would help to prevent blowing past the API limits so I'll probably implement that soon.

I've also got a private branch that's a work in progress to make the pydora output back-ends pluggable. I'm doing some testing with VLC in headless mode that looks promising but I don't really have an ETA on that yet.

I'm going to try to repro this specific issue again so will leave this ticket open in case I find something.
